lib/ecto/query/builder.ex

```diff
@@ -168,6 +168,10 @@ defmodule Ecto.Query.Builder do
   end
 
   # subqueries
+  def escape({:subquery, _, _} = expr, type, {params, {_take, subqueries}}, vars, env) do
+    escape(expr, type, {params, subqueries}, vars, env)
```

Comment:
This is the code that is making the `take` disappear. It is probably best to introduce a function called `escape_subquery` which can be used by this function and the one below, so you do something like this:

```elixir
  def escape({:subquery, _, _} = expr, type, {params, {take, subqueries}}, _vars, _env) do
    {expr, params, subqueries} = escape_subquery(expr, params, subqueries)
    {expr, {params, {take, subqueries}}}
```